src/inmanta/logging.py

```diff
@@ -554,15 +591,18 @@ def apply_options(self, options: Options) -> None:
         )
         self._handler = self._apply_logging_config(logging_config)
 
-    def _apply_logging_config(self, logging_config: FullLoggingConfig) -> logging.Handler:
+    def _apply_logging_config(self, logging_config: FullLoggingConfig) -> Optional[logging.Handler]:
         """
         Apply the given logging_config as the current configuration of the logging system.
 
         This method assume that the given config defines a single root handler.
         """
         logging_config.apply_config()
-        assert len(logging.root.handlers) == 1
-        return logging.root.handlers[0]
+        if inmanta.RUNNING_TESTS and CAPLOG_FIXTURE_USED:
```

Comment:
Proposal to lift the caplog logic from the source to the tests. Feel free to leave it as is if you prefer though.
1. make `self._handler: Optional[logging.Handler]` -> `self._handlers: Sequence[logging.Handler]`, and have this method return `list[logging.Handler]`.
2. In `set_log...` methods, just iterate over the handlers (which may be an empty list)
3. Move the `assert len(...)` to `self.get_handler` implementation. It is the only place where this is still relevant (I think), and it's deprecated.
4. monkeypatch `FullLoggingConfig._to_dict_config` instead of the `CAPLOG_FIXTURE_USED` check in `FullLoggingConfig.apply_config`.